addons/l10n_tr_nilvera_einvoice/wizard/account_move_send.py

```diff
@@ -38,22 +35,32 @@ def _get_default_l10n_tr_nilvera_einvoice_enable_einvoice(self, move):
         return move.l10n_tr_nilvera_send_status == 'not_sent' and move.is_invoice(include_receipts=True) and move.country_code == 'TR'
 
     def _l10n_tr_nilvera_check_invoices(self):
+        error_messages = {}
         moves_to_check = self.move_ids.filtered(self._get_default_l10n_tr_nilvera_einvoice_enable_einvoice)
         invalid_records = moves_to_check.partner_id.filtered(
             lambda p: p.country_code != 'TR' or not p.city or not p.state_id or not p.street
         )
+        invalid_subscription_dates = moves_to_check.filtered(
+            lambda move: move._l10n_tr_nilvera_einvoice_check_xml_subscription_dates()
+        )
+        if invalid_subscription_dates:
+            error_messages["invalid_subscription_dates"] = {
+                "message": _("Please make sure all the products have same start date and end date in following Invoice(s)"),
```

Comment:
A bit weird to talk about the products having a start and end date.
Why not Lines/ Invoice Lines?